src/context/FoxEthProvider/FoxEthProvider.tsx

```diff
@@ -180,45 +197,82 @@ export const FoxEthProvider = ({ children }: FoxEthProviderProps) => {
         })
       })
     })()
-  }, [accountAddress, ethAccountIds, dispatch, foxFarmingEnabled, readyToFetchFarmingData])
+  }, [ethAccountIds, dispatch, foxFarmingEnabled, readyToFetchFarmingData])
 
   const transaction = useAppSelector(gs => selectTxById(gs, ongoingTxId ?? ''))
 
   const handleOngoingTxIdChange = useCallback(
-    async (txid: string, contractAddress?: string) => {
-      if (!accountAddress) return
+    async (type: 'farming' | 'lp', txid: string, contractAddress?: string) => {
+      const accountId = type === 'farming' ? farmingAccountId : lpAccountId
+      const accountAddress = type === 'farming' ? farmingAccountAddress : lpAccountAddress
       setOngoingTxId(serializeTxIndex(accountId ?? '', txid, accountAddress))
       if (contractAddress) setOngoingTxContractAddress(contractAddress)
     },
-    [accountId, accountAddress],
+    [lpAccountId, lpAccountAddress, farmingAccountId, farmingAccountAddress],
```

Comment:
This will invalidate on any of the LP/farming deps changes, which is not optimal. We could remove the `handleOngoingTxIdChange` dependency on the curried methods if that's an issue